R/appMetadata.R

```diff
@@ -7,6 +7,7 @@ appMetadata <- function(appDir,
                         metadata = list()) {
 
   appFiles <- standardizeAppFiles(appDir, appFiles)
+  checkAppLayout(appDir, appPrimaryDoc)
```

Comment:
Moving this "lint-check" here means that we have already created a deployment target on the remote server. If this check is firing, that means that folks most likely are trying to deploy the wrong thing; we've just created something that they will need to remove on the remote server.

Reply:
Ah, good catch. Then let's maybe leave this to mature for a while, until `appMetadata()` is computed before we create the deployment target. (Which I think I'm gradually moving towards anyway.)

Reply:
Can you mark this as "needs changes" so I don't get over zealous and move it?